### ISSUE TITLE
[Discussion] about preprocessor definitions for riscv64

### ISSUE THREAD

**Issue by unicornx** [riscv-android-src/platform-bionic#10]
riscv64-android-12.0.0_dev for bionic

I find current code use two methods to flag rv64, for example:

```
#elif __riscv_xlen == 64
......
```
or
```
(defined(__riscv) && (__riscv_xlen == 64)
```

I suggest we should unify the coding style. Which one do you prefer? First one or the second option?

**Comment by guoyinchen** [comment]
Good suggestion. In current build option(only riscv64 been supported), they are same impact. 
I think the unifed coding style should be below. @MaoHan001 , what is your idea?
(defined(__riscv) && (__riscv_xlen == 64)

**Comment by MaoHan001** [comment]
glibc normally use a single __riscv_xlen == 64 to detect target length for files under the riscv fold(why we use this style before).  And the most file require these macro are non-architecture specific under bionic. So I think it is better to use:
(defined(__riscv) && (__riscv_xlen == 64)
or
(defined(__riscv) && (\_\_LP64\_\_)

**Comment by unicornx** [comment]
Thanks, I prefer "(defined(__riscv) && (__riscv_xlen == 64)", because "__riscv" and "__riscv_xlen == 64" are both defined in <https://github.com/riscv-non-isa/riscv-toolchain-conventions#cc-preprocessor-definitions>

If all of you agreed, I would create a PR and change it.

**Comment by MaoHan001** [comment]
> Thanks, I prefer "(defined(__riscv) && (__riscv_xlen == 64)", because "__riscv" and "__riscv_xlen == 64" are both defined in https://github.com/riscv-non-isa/riscv-toolchain-conventions#cc-preprocessor-definitions
> 
> If all of you agreed, I would create a PR and change it.

Yes, Please : )

**Comment by unicornx** [comment]
fixed with pr #13 

close this issue.
